src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -277,6 +277,10 @@ public static Runspace CreateRunspace(PSHost psHost)
                 initialSessionState = InitialSessionState.CreateDefault2();
             }
 
+            // Create and initialize a new Runspace while honoring the LanguageMode.
+            Console.WriteLine(languageMode);
+            initialSessionState.LanguageMode = languageMode;
```

Comment:
Have we confirmed this is all that's done in `ConsoleHost`?  I wonder if it sets up whitelists as well maybe?

Not a blocking comment though. Having this properly propagate the correct language mode is first priority, any extras can come later imo.

Reply:
I talked with @TravisEz13 @PaulHigin and @daxian-dbw on this. This was the recommended way to go. Paul pointed me to places in Jobs where it queries for the LanguageMode (via a PowerShell API that's only on Windows) and sets in in the ISS. Then Dongbo suggested we just inherit like this and everyone agreed.

Reply:
We are making this language mode aware, so I would like to see more comments about why this code is here (maybe where LanguageMode property is defined).  My concern is that future maintainers will not understand this code, and not maintain it correctly.